autoload/scnvim/editor.vim

```diff
@@ -0,0 +1,69 @@
+" File: scnvim/autoload/editor.vim
+" Author: David Granström
+" Description: scnvim editor helpers
+" Note: Will be deprecated and moved to editor.lua in the future.
+
+scriptencoding utf-8
+
+function! s:skip_pattern() abort
+  return 'synIDattr(synID(line("."), col("."), 0), "name") ' .
+        \ '=~? "scLineComment\\|scComment\\|scString\\|scSymbol"'
+endfunction
+
+function! s:find_match(start, end, flags) abort
+    return searchpairpos(a:start, '', a:end, a:flags, s:skip_pattern())
+endfunction
+
+function! scnvim#editor#get_block() abort
+    " initialize to invalid ranges
+    let start_pos = [0, 0]
+    let end_pos = [0, 0]
+    let forward_flags = 'nW'
+    let backward_flags = 'nbW'
+    " searchpairpos starts the search from the cursor so save where we are
+    " right now and restore the cursor after the search
+    let c_curpos = getcurpos()
+    " move to first column
+    call setpos('.', [0, c_curpos[1], 1, 0])
+    let [xs, ys] = s:find_match('(', ')', backward_flags)
+    let start_pos = [xs, ys]
+    if xs == 0 && ys == 0
+      " we are already standing on the opening brace
+      let start_pos = [line('.'), col('.')]
+    else
+      while xs > 0 && ys > 0
+        call setpos('.', [0, xs, ys, 0])
+        let start_pos = [xs, ys]
+        let [xs, ys] = s:find_match('(', ')', backward_flags)
+      endwhile
+    endif
+    call setpos('.', [0, start_pos[0], start_pos[1], 0])
+    let end_pos = s:find_match('(', ')', forward_flags)
+    " restore cursor
+    call setpos('.', c_curpos)
+    return [start_pos[0], end_pos[0]]
+endfunction
+
+" Could be replaced with `vim.region` later
+function! scnvim#editor#get_visual_selection() abort
```

Comment:
Just an FYI, if you are keeping these purely for the lack of get_visual_range in lua, I use this in my rc
```lua
function utils.get_visual_selection(bufnr)
    vim.api.nvim_input "<Esc>"
    vim.api.nvim_input "gv"
    local first = vim.api.nvim_buf_get_mark(bufnr, "<")
    local last = vim.api.nvim_buf_get_mark(bufnr, ">")
    local lines = vim.api.nvim_buf_get_lines(bufnr, first[1], last[1], true)

    return {
        lines = lines,
        line_start = first[1],
        line_end = last[1],
        col_start = first[2],
        col_end = last[2],
    }
end
```

Reply:
(sorry, my previous comment was apparently not saved before marking this as resolved)

Thanks! These two functions will be converted to lua eventually, I'd just like to get a few more unit test for the editor module before doing the refactoring. I'm planning to do a follow up PR for this after this one has been is merged.